src/dparse/trivia.d

```diff
@@ -0,0 +1,696 @@
+/**
+ * Module to work with trivia tokens (`comment`, `whitespace`,
+ * `specialTokenSequence`) which are attached to tokens near them when source
+ * code gets tokenized.
+ */
+module dparse.trivia;
+
+import std.algorithm;
+import std.array;
+import std.range;
+import std.string;
+import std.traits;
+
+import dparse.lexer;
+
+enum CommentType : ubyte
+{
+    none,
+    docLine,
+    docBlock,
+    normalLine,
+    normalBlock,
+}
+
+CommentType determineCommentType(string comment) pure nothrow @safe
+{
+    auto bytes = comment.representation;
+    auto index = bytes.startsWith(
+        "//".representation,
+        "/+".representation,
+        "/*".representation
+    );
+    bool isDoc = bytes.length >= 3 && bytes[1] == bytes[2];
+    switch (index)
+    {
+    case 1:
+        return isDoc ? CommentType.docLine :  CommentType.normalLine;
+    case 2:
+    case 3:
+        return isDoc ? CommentType.docBlock :  CommentType.normalBlock;
+    default:
+        return CommentType.none;
```

Comment:
Is is ever valid for this to happen? How would the lexer create a comment token that didn't start with one of the expected character sequences?

Reply:
not necessarily happening by the lexer, but rather when you use this function outside of analyzing lexer output (like user input)

Anyway I think it's better than an assert false if this case does ever occur.